tests_integration/runPrettier.js

```diff
@@ -82,9 +83,9 @@ function runPrettier(dir, args, options) {
   // We cannot use `jest.setMock("get-stream", impl)` here, because in the
   // production build everything is bundled into one file so there is no
   // "get-stream" module to mock.
-  jest.spyOn(require(thirdParty), "getStream").mockImplementation(() => ({
-    then: handler => handler(options.input || "")
```

Comment:
This was a hacky mock to make the promise returned by `getStream` synchronous. That _mostly_ worked fine until I tried to chain a `.catch` on it (see above), which failed with `TypeError: Cannot read property 'catch' of undefined`. 

Why did I say _mostly?_ Because `.then(callback)` is supposed to run `callback()` in a `try-catch` and reject the promise if `callback()` throws. This resulted in config errors not being caught by the regular error handler but by the `try-catch` below in this file! This uncovered a bug where we printed stack traces for config errors which looks ugly.

Before:

```
❯ prettier --parser babel < test.js
(node:21531) UnhandledPromiseRejectionWarning: Error: Invalid printWidth value. Expected an integer, but received "nope".
    at _loop (/home/lydell/forks/prettier/node_modules/prettier/bin-prettier.js:7887:63)
    at Normalizer._applyNormalization (/home/lydell/forks/prettier/node_modules/prettier/bin-prettier.js:8000:13)
    at applyNormalization (/home/lydell/forks/prettier/node_modules/prettier/bin-prettier.js:7817:49)
    at Normalizer.normalize (/home/lydell/forks/prettier/node_modules/prettier/bin-prettier.js:7823:9)
    at normalizeOptions$1 (/home/lydell/forks/prettier/node_modules/prettier/bin-prettier.js:8760:31)
    at Object.normalizeApiOptions (/home/lydell/forks/prettier/node_modules/prettier/bin-prettier.js:8918:10)
    at getOptionsForFile (/home/lydell/forks/prettier/node_modules/prettier/bin-prettier.js:44160:69)
    at /home/lydell/forks/prettier/node_modules/prettier/bin-prettier.js:44214:22
    at process._tickCallback (internal/process/next_tick.js:68:7)
(node:21531) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:21531) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

```

After:

```
❯ bin/prettier.js --parser babel < test.js
[error] Invalid printWidth value. Expected an integer, but received "nope".
```

Reply:
haha